config/v1/types_infrastructure.go

```diff
@@ -1417,6 +1420,85 @@ type NutanixPlatformSpec struct {
 	// +listType=map
 	// +listMapKey=name
 	PrismElements []NutanixPrismElementEndpoint `json:"prismElements"`
+
+	// failureDomains configures failure domains information for the Nutanix platform.
+	// If this is omitted, failure domains will not be used to spread Machines.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	FailureDomains []NutanixFailureDomain `json:"failureDomains"`
```

Comment:
The failure domains are defined in the Infrastructure resource Nutanix providerSpec. The CPMS and MachineSet CRs will hold references (by name) to the failure domains defined if they need to spread Machines across failure domains. User can add new failure domain items to or remove items from the list in the Infrastructure resource. It will affect the existing CPMS CR if it has references to the failure domain items removed or name changed. In this case, the CPMSO will report errors.

Reply:
Have you tested this flow? For example, I'd be interested to see what the errors look like when you remove a failure domain from the infrastructure object, and then CPMS tries to reconcile. I guess it will not know what to do with the Machine?

Reply:
@JoelSpeed I just did the test to remove a failureDomain named "fd1" from the Infrastructure CR in my test cluster, and the below error appears in the CPMSO logs:

E1117 14:29:29.223760       1 controller.go:329]  "msg"="Reconciler error" "error"="error reconciling control plane machine set: error fetching machine info: could not generate machine info for machine cluster-g3-j2wrr-master-hk77m-1: error injecting failure domain into provider config: failed to inject failure domain: unknown failure domain: The failure domain with name \"fd1\" is not defined in the infrastructure resource" "controller"="controlplanemachineset" "reconcileID"="028e853c-7404-45b4-970b-ad93a6efbbf7"